pkg/record/record.go

```diff
@@ -147,6 +148,19 @@ func (r Record) mapData(d Data) interface{} {
 	return nil
 }
 
+func (r Record) Clone() (Record, error) {
+	clone := Record{}
+	err := copier.CopyWithOption(
```

Comment:
I have an issue with using a library with reflection for this 😕 It's a function that will be called at least once for every record flowing through the pipeline. I don't have benchmarks to prove it, but I imagine this being far slower than just manually copying fields (including deep copying maps). Sure, it's easier with this library, but since this is the hot path I wouldn't want to take shortcuts here.

Reply:
Yeah, I considered that as well. My first version was in fact copying fields "manually". However, the after and before fields can be `StructuredData` which is `map[string]interface{}`. Implementing a deep copy option for `interface{}` would take some time, so for this PR I chose to first get the functionality fixed (especially because it's easy to reproduce) versus optimizing (for high volume streams).

Reply:
I see, good point, `interface{}` does indeed make things harder, it's not a trivial thing to get right, we'll need to recursively traverse it if it's a map or a slice. And we'll need to extensively test this for all types. You convinced me, let's do this separately, but we should not forget about it, please create an issue for this 🙏

Regarding that issue - let's consider what can be in that `interface{}`. We get that from the connector so it can't just be _anything_, it has to be something that gRPC can handle - it's either a basic type, a slice or a map (did I forget anything?). We plan to add [`time.Time`](https://github.com/ConduitIO/conduit/issues/866) to the list of supported types, but that's also a type that's used as a value and easily copied. So sounds like if we can copy a map with these values we should be good, and that should be doable without reflection.

Reply:
Forgot to mention the issue for the optimizing record cloning: https://github.com/ConduitIO/conduit/issues/885.

Reply:
> Regarding that issue - let's consider what can be in that `interface{}`. We get that from the connector so it can't just be _anything_, it has to be something that gRPC can handle - it's either a basic type, a slice, or a map (did I forget anything?).

IIUC, the built-in connectors, in theory, can return pretty much anything, but the Conduit protocol is based on gRPC, so we need to be aligned with it?

Reply:
That's right, we need to design for the common denominator, so gRPC and standalone connectors.